src/components/Company/PaySchedule/PaySchedule.tsx

```diff
@@ -0,0 +1,361 @@
+      auto_pilot: defaultValues?.auto_pilot?.toString() ?? '',
+      custom_twice_per_month: 'false',
+    }
+  }, [defaultValues])
+
+  const [payScheduleDraft, setPayScheduleDraft] = useState<PaySchedulePreviewDraft | null>(null)
+  const {
+    data: payPreviewData,
+    error: paySchedulePreviewError,
+    isLoading,
+  } = useGetPaySchedulePreview(
+    companyId,
+    payScheduleDraft as PaySchedulePreviewDraft, // Casting to non-null because we know it's not null from the enabled prop
+    !!payScheduleDraft,
+  )
+
+  const { data: paySchedules } = useGetAllPaySchedules(companyId)
+
+  const createPayScheduleMutation = useCreatePaySchedule()
+  const updatePayScheduleMutation = useUpdatePaySchedule()
+
+  const formMethods = useForm<PayScheduleInputs, unknown, PayScheduleOutputs>({
+    resolver: valibotResolver(PayScheduleSchema),
+    defaultValues: transformedDefaultValues,
+  })
+
+  const allValues = formMethods.watch()
+
+  // Set the custom_twice_per_month value based on the frequency and day_1 and day_2 values as it is not set by the API call
+  useEffect(() => {
+    if (
+      allValues.frequency === 'Twice per month' &&
+      allValues.day_1 === 15 &&
+      allValues.day_2 === 31 &&
+      allValues.custom_twice_per_month === undefined
+    ) {
+      formMethods.setValue('custom_twice_per_month', `false`)
+    } else if (
+      allValues.frequency === 'Twice per month' &&
+      allValues.custom_twice_per_month === undefined
+    ) {
+      formMethods.setValue('custom_twice_per_month', `true`)
+    }
+  }, [
+    allValues.frequency,
+    allValues.day_1,
+    allValues.day_2,
+    formMethods,
+    allValues.custom_twice_per_month,
+  ])
+
+  useEffect(() => {
+    // Don't update if dates are not set
+    if (!allValues.anchor_pay_date || !allValues.anchor_end_of_pay_period) {
+      return
+    }
+
+    setPayScheduleDraft({
+      frequency: allValues.frequency as PaySchedulePreviewDraft['frequency'],
+      anchor_pay_date: allValues.anchor_pay_date.toString(),
+      anchor_end_of_pay_period: allValues.anchor_end_of_pay_period.toString(),
+      day_1: allValues.day_1 || undefined,
+      day_2: allValues.day_2 || undefined,
+    })
+  }, [
+    allValues.anchor_end_of_pay_period,
+    allValues.anchor_pay_date,
+    allValues.day_1,
+    allValues.day_2,
+    allValues.frequency,
+    setPayScheduleDraft,
+  ])
+
+  // Custom effect to show/hide pay schedule preview errors
+  useEffect(() => {
+    if (paySchedulePreviewError?.message) {
+      const errors: {
+        payload: {
+          error_key: string
+          message: string
+        }[]
+      } = JSON.parse(paySchedulePreviewError.message)
+      const errorsList = errors.payload
+
+      errorsList.forEach(error => {
+        formMethods.setError(error.error_key as keyof PayScheduleInputs, { message: error.message })
+      })
+    } else {
+      formMethods.clearErrors()
+    }
+  }, [formMethods, paySchedulePreviewError])
```

Comment:
Same here, rec to keep `formMethods` out of the dep array and instead just include `clearErrors`

Reply:
Resolved